### DISCUSSION TITLE
CLI option to run arbitrary command

### DISCUSSION THREAD

**Discussion by kanashimia** [helix-editor/helix#10021] [Ideas]
https://github.com/helix-editor/helix/pull/8521 added +N number handling to helix, but in vim +N actually executes command N.
Helix already handles :N so it would be possible to generalise the behaviour.

Possible flag names are: `--cmd` `-C` `-x` `-e` `--run` `--exec`
Possible description:
```
    -C, --cmd <command>            Specifies a command to run at startup, can be passed multiple times
```

This flag will be useful for scripting. Another one could be added later to source commands from 

There are two ways it could work:
1.  It could either allow only named commands, as in [helix-term/src/commands.rs  static_commands!](https://github.com/helix-editor/helix/blob/f21efc7a2fe579d06867d63c3ec86b1458415707/helix-term/src/commands.rs#L231)
2. It could could work like a key sequence input.

Exiting immediately using these two approaches:
1. `hx --cmd quit`
2. `hx --cmd ':quit<ret>'`

It is also possible to keep + for the first approach and and add --cmd for the second one.

---

I tried to implement the second approach like this:
```rs
for command in args.commands {
    for key_event in helix_view::input::parse_macro(&command).unwrap() {
        app.compositor.handle_event(
            &Event::Key(key_event),
            &mut cx
        );
    }
}
```

That code is in `Application::new` just after `let app =`, but it seems to break on `:quit<ret>`
```
› cargo run -- -C ':quit<ret>'  
thread 'main' panicked at helix-view/src/tree.rs:278:29:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
I believe it requires view or something to be already created, couldn't figure out how to do that without adding a new event type, as that seemed too complicated for no reason, so for now will just keep this issue here.
I also didn't find an API to implement the first approach.

**Comment by the-mikedavis** [comment]
This was discussed a little in the initial PR that lead to #8521 https://github.com/helix-editor/helix/pull/5603#discussion_r1083327983 and I believe elsewhere but I can't find the link. I'm very skeptical that we need this feature. Helix is meant to be run interactively so we try to keep the argument parsing and behavior very simple.
